moveit_ros/planning/planning_request_adapter_plugins/src/publish_intermediate_trajectory.cpp

```diff
@@ -0,0 +1,81 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2012, Willow Garage, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Bianca Homberg, based off empty.cpp by Ioan Sucan */
+
+#include <moveit/planning_request_adapter/planning_request_adapter.h>
+#include <class_loader/class_loader.hpp>
+
+namespace default_planner_request_adapters
+{
+class PublishIntermediateTrajectory : public planning_request_adapter::PlanningRequestAdapter
+{
+public:
+  ros::Publisher intermediate_trajectory_;
+  ros::NodeHandle nh_;
+
+  PublishIntermediateTrajectory() : planning_request_adapter::PlanningRequestAdapter(), nh_("~")
+  {
+  }
+
+  void initialize(const ros::NodeHandle& node_handle) override
+  {
+    std::string topic_name = "intermediate_trajectory";
+    node_handle.getParam("intermediate_trajectory_topic_name", topic_name);
+    intermediate_trajectory_ = nh_.advertise<moveit_msgs::RobotTrajectory>(topic_name, 1000);
+  }
+
+  std::string getDescription() const override
+  {
+    return "Publish intermediate trajectory.";
```

Comment:
```suggestion
    return "Publish intermediate trajectory between planning request adapters.";
```

Reply:
What if this is the only planning request adapter loaded? :slightly_smiling_face:

Reply:
It takes the current value of the MotionPlanResponse and reads out the trajectory stored in it.  It isn't required to be in any particular location within the planning adapter chain.

Changing to "Publish intermediate trajectory at current location in planning adapter chain", unless either of you has a clearer suggestion!